File: Virtual Machine/Nova.VirtualMachine/Nova.VirtualMachine/ByteMethod.cpp
#include "ByteClass.h"
#include "ByteMethod.h"


void ByteMethod::Deserialize(BinaryReader& reader)
{
	this->name = reader.ReadString();
	this->modifiers = (Modifiers)reader.Read<char>();
	this->parametersCount = reader.Read<int>();

	this->block = new ByteBlock();
	this->block->Deserialize(reader);
}

ByteMethod::ByteMethod(ByteClass* parent)
{
	this->parent = parent;
}

void ByteMethod::Dispose()
{
	delete block;
}

File: Compiler/Nova.VirtualMachine/Members/ByteMethod.cs
using Nova.Utils.IO;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Nova.VirtualMachine.Members
{
    public class ByteMethod
    {
        public string Name
        {
            get;
            private set;
        }
        public ByteClass Parent
        {
            get;
            private set;
        }
        public Modifiers Modifiers
        {
            get;
            private set;
        }
        public int ParametersCount
        {
            get;
            private set;
        }
        public ByteBlock Block
        {
            get;
            private set;
        }
        public ByteMethod(ByteClass parent)
        {
            this.Parent = parent;
            this.Block = new ByteBlock();
        }
        public void Deserialize(CppBinaryReader reader)
        {
            this.Name = reader.ReadString();
            this.Modifiers = (Modifiers)reader.ReadByte();
            this.ParametersCount = reader.ReadInt32();

            this.Block.Deserialize(reader);
        }
    }
}
